src/ome2024_ngff_challenge/resave.py

```diff
@@ -856,16 +527,105 @@ def main(ns: argparse.Namespace, rocrate: ROCrateWriter | None = None) -> int:
     else:
         LOGGER.warning(f"no convertible metadata: {input_config.zr_attrs.keys()}")
 
+    if converted == 0:
+        raise SystemExit(1)
     return converted
 
 
-def cli(args=sys.argv[1:]):
+def cli(subparsers: argparse._SubParsersAction):
     """
     Parses the arguments contained in `args` and passes
     them to `main`. If no images are converted, raises
     SystemExit. Otherwise, return the number of images.
     """
-    parser = argparse.ArgumentParser()
+    cmd = "ome2024-ngff-challenge resave"
+    desc = f"""
+
+The `resave` subcommand will convert an existing Zarr v2 dataset into
+a Zarr v3 dataset according to the challenge specification. Additionally,
+a number of options are available for adding metadata
+
+
+BASIC
+
+Simplest example:                        {cmd} --cc-by in.zarr out.zarr
+Overwrite existing output:               {cmd} --cc-by in.zarr out.zarr --output-overwrite
+
+
+METADATA
+
+There are three levels of metadata that the challenge is looking for:
+
+    - strongly recommended: license
+    - recommended: organism and modality
+    - optional: name and description
+
+Most suggested license                   {cmd} in.zarr out.zarr --cc-by
+Public domain license                    {cmd} in.zarr out.zarr --cc0
+Choose your own license                  {cmd} in.zarr out.zarr --rocrate-license=https://creativecommons.org/licenses/by-nc/4.0/
+
+Organism: Arabidopsis thaliana           {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid3702
+Organism: Drosophila melanogaster        {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid7227
+Organism: Escherichia coli               {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid562
+Organism: Homo sapiens                   {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid9606
+Organism: Mus musculus                   {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid10090
+Organism: Saccharomyces cerevisiae       {cmd} in.zarr out.zarr --cc0 --rocrate-organism=NCBI:txid4932
+
+Modality: bright-field microscopy        {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000243
+Modality: confocal microscopy            {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000251
+Modality: two-photon laser scanning      {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000253
+Modality: two-photon laser scanning      {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000253
+Modality: scanning electrom microscopy   {cmd} in.zarr out.zarr --cc0 --rocrate-modality=obo:FBbi_00000257
```

Comment:
🤷🏽 cc: @AybukeKY @sherwoodf

Reply:
The latter is easiest and can definitely be the starting point. I know I suggested the look-up but I guess sticking to the fbbi may also help us somewhat avoid/ignore the wider problem that not everything (e.g. new tech) is on fbbi.

Reply:
Well verifying that it starts with obo:FBbi_ won't solve fbbi not being up to date, but since it's a taxonomy i'd expect there to be _some_ generic type that's relevant, even if it's not the most specific one a user would like to have.

I think validating that it starts with obo:FBbi or i guess `http://purl.obolibrary.org/obo/FBbi_` would probably cover most cases/push people enough to actually go look up values.